kaggle_environments/core.py

```diff
@@ -196,7 +196,7 @@ def step(self, actions, logs=[]):
         self.state = self.__run_interpreter(action_state, logs)
 
         # Max Steps reached. Mark ACTIVE/INACTIVE agents as DONE.
-        if len(self.steps) == self.configuration.episodeSteps - 1:
```

Comment:
Historically, the `observation.step` field was implemented on a per-environment basis which led to inconsistent step implementations. Because step is useful for all discrete environments it's now handled automatically by the framework. Accordingly in places where we previously calculated the current step via `len(self.steps)` we can now retrieve the step directly from the first agent's observation (step only appears on the first agent because it's a shared property and is the same for all agents).

There is a corresponding change to the observation schema that adds this property to all existing environment observations: https://github.com/Kaggle/kaggle-environments/pull/116/files#diff-908247f9111290a149603cca59ee046029770d58119d9176df4dd578fd6eeed6R110